### PR TITLE
Profiles/bug fix

### PR DESCRIPTION
**Pull request by Thosam1**
some events have users do not exists, which in turn leads to having a blank case in the list of participants, 

I made it so that in the profileFirebaseDataSource we only return the profile if it is not equal to null

### CODE REVIEW

File: app/src/androidTest/java/com/sdpteam/connectout/profile/ProfileFirebaseDataSourceTest.java [hunk 1]
```diff
@@ -44,14 +44,6 @@ public void fetchesAllGivenProfiles() {
...
     }
 
-    @Test
-    public void fetchNullProfileWithWrongId() {
-        ProfileFirebaseDataSource model = new ProfileFirebaseDataSource();
-        Profile p = fJoin(model.fetchProfile(UUID.randomUUID().toString() + "NotAPossibleId"));
-
-        assertNull(p);
-    }
-
```
Comment (EricSaikali) [inline app/src/androidTest/java/com/sdpteam/connectout/profile/ProfileFirebaseDataSourceTest.java:54]: The deletion of this test surprised me, could you explain why did you delete it ? I have trouble understanding why its not relevant anymore

File: app/src/androidTest/java/com/sdpteam/connectout/profile/ProfileTest.java [hunk 2]
```diff
@@ -143,6 +143,5 @@ public void deleteProfileTest() {
         fJoin(model.saveProfile(userProfile));
         model.deleteProfile(uid);
         waitABit();
-        assertNull(fJoin(model.fetchProfile(uid)));
```
Comment (EricSaikali) [inline app/src/androidTest/java/com/sdpteam/connectout/profile/ProfileTest.java:146]: Same here, I do not understand why this assertion was removed

File: app/src/androidTest/java/com/sdpteam/connectout/profileList/CommunityActivityTest.java [hunk 3]
```diff
@@ -90,10 +97,11 @@ public void clickingOnAUserLaunchesRightProfilePage() {
...
 
         String expectedUid = (p == null) ? NULL_USER : list.get(userIndexToCheck).getId();
+        System.out.println(expectedUid);
 
-/*TODO fix ProfileList bug
-        onView(withIndex(withId(R.id.nameAdapterTextView), userIndexToCheck)).perform(click());
-        intended(Matchers.allOf(hasComponent(ProfileActivity.class.getName()), hasExtra(equalTo("uid"), equalTo(expectedUid))));*/
+        //TODO fix ProfileList bug
+//        onView(withIndex(withId(R.id.nameAdapterTextView), userIndexToCheck)).perform(click());
+//        intended(Matchers.allOf(hasComponent(ProfileActivity.class.getName()), hasExtra(equalTo("uid"), equalTo(expectedUid))));
```
Comment (EricSaikali) [inline app/src/androidTest/java/com/sdpteam/connectout/profileList/CommunityActivityTest.java:104]: I believe this TODO is one that is treated through this task, why does it remain here ?

**Review by EricSaikali — COMMENTED** [review]
Good job with the functionalities !
Please answer me on the few question I have on the different technical changes you made.

Nice work

File: app/src/androidTest/java/com/sdpteam/connectout/profile/ProfileTest.java (hunk 2 above)
Comment (Thosam1) [reply app/src/androidTest/java/com/sdpteam/connectout/profile/ProfileTest.java:146]: because the fjoin will never return something since the uid doesn't even exist

Comment (Thosam1) [reply app/src/androidTest/java/com/sdpteam/connectout/profile/ProfileTest.java:146]: previously, it returns null, and then we don't render on the view depending on whether the profile is null or not. However, we don't even want the null profiles to be in the datasource

File: app/src/androidTest/java/com/sdpteam/connectout/profile/ProfileFirebaseDataSourceTest.java (hunk 1 above)
Comment (Thosam1) [reply app/src/androidTest/java/com/sdpteam/connectout/profile/ProfileFirebaseDataSourceTest.java:54]: Like look at the fetchProfiles method : 

List<Profile> profiles = new ArrayList<>();
        allTasks.addOnCompleteListener(dataSnapshots -> {
            for (DataSnapshot dataSnapshot : dataSnapshots.getResult()) {
                Profile profile = dataSnapshot.getValue(Profile.class);
                if(profile != null) {
                    profiles.add(profile);
                }
            }
            futures.complete(profiles);
        });
        
Inside, the problem is that if we have a list of 7 users with incorrect ids, and 3 with correct ids, the profiles returned would be a list of 7 null values and 3 profile objects

Comment (EricSaikali) [reply app/src/androidTest/java/com/sdpteam/connectout/profile/ProfileFirebaseDataSourceTest.java:54]: Thanks for your explication !

File: app/src/androidTest/java/com/sdpteam/connectout/profile/ProfileTest.java (hunk 2 above)
Comment (EricSaikali) [reply app/src/androidTest/java/com/sdpteam/connectout/profile/ProfileTest.java:146]: This is much clearer now, thanks.

**Review by EricSaikali — APPROVED** [review]
Good job!
